src/lib/annotations/AnnotationThread.js

```diff
@@ -146,29 +146,10 @@ class AnnotationThread extends EventEmitter {
 
         // Changing state from pending
         this.state = STATES.hover;
-
         // Save annotation on server
         this.annotationService
             .create(annotationData)
-            .then((savedAnnotation) => {
-                // If no temporary annotation is found, save to thread normally
-                const tempIdx = this.annotations.indexOf(tempAnnotation);
-                if (tempIdx === -1) {
-                    this.saveAnnotationToThread(savedAnnotation);
-                }
-
-                // Add thread number to associated dialog and thread
-                this.threadNumber = this.threadNumber || savedAnnotation.threadNumber;
-                this.dialog.element.dataset.threadNumber = this.threadNumber;
-
-                // Otherwise, replace temporary annotation with annotation saved to server
-                this.annotations[tempIdx] = savedAnnotation;
-
-                if (this.dialog) {
-                    this.dialog.addAnnotation(savedAnnotation);
-                    this.dialog.removeAnnotation(tempAnnotationID);
-                }
-            })
+            .then((savedAnnotation) => this.updateLocalAnnotationFromServer(savedAnnotation, tempAnnotation))
```

Comment:
I'd like to keep the naming consistent. Can we make this method either updateTempAnnotation(...) or change tempAnnotation, tempIdx, and other temp variables to localAnnotation, etc?